posthog/clickhouse/client/limit.py

```diff
@@ -155,6 +167,32 @@ def use(self, *args, **kwargs) -> tuple[Optional[str], Optional[str]]:
 
         return running_tasks_key, task_id
 
+    def _get_org_concurrency_limit(self, org_id: int) -> Optional[int]:
```

Comment:
the implementation of this function is great (especially with the redis caching of the feature limit)
though I'd prefer if we keep this function separate from `RateLimit` class. Maybe the rate limit class can accept a callable for max_concurrency. When the arg is a callable and not an int, then it uses the callable to determine max concurrency?

Reply:
@anirudhpillai decided against the callable because it made it move convoluted that it needed to be, but I did move it out of the rate limit class.